### ISSUE TITLE
get_project 'root' keyword fails silently

### ISSUE THREAD

**Issue by rskye144** [glotzerlab/signac#779]
### Description
When the `get_project` function is called with a project root that does not exist, it silently defaults to a project in the current directory. 

### To reproduce

```python
import signac
project1 = signac.init_project()
project2 = signac.get_project("some/directory/that/does/not/exist")  # this should raise to prevent user error, rather than searching upward from a nonexistent directory
assert project1 == project2  # this is true but we shouldn't get this far
```

Error also occurs because paths with `~` are not expanded, producing an unexpected error; @bdice suggests could consider adding:

```python
if "~" in root:
    warnings.warn("The character '~' is in the provided root path. Did you mean to call os.path.expanduser(root)?", UserWarning)
```

### Error output

No error is raised (which is wrong, an error/warning should be raised)

**Comment by mikemhenry** [comment]
@rskye144 :wave: thanks for the bug report :heart_eyes: 

Are you interested in contributing a PR to fix this issue? We can provide some help/guidance if you are interested!

**Comment by rskye144** [comment]
I can give a PR a try! It will be my first time doing that, so any advice is appreciated.

**Comment by bdice** [comment]
One key piece that I didn't notice earlier: this bug occurs fails for _relative paths contained inside the current directory_, when the current directory contains a signac project. Getting a nonexistent absolute path like `signac.get_project("/a/fake/path")` fails with a `LookupError`.

**Comment by vyasr** [comment]
Resolved by #792
